error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/CanonicalConstantFieldName.java

```diff
@@ -0,0 +1,184 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.errorprone.BugPattern.LinkType.CUSTOM;
+import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
+import static com.google.errorprone.BugPattern.StandardTags.LIKELY_ERROR;
+import static com.google.errorprone.matchers.Matchers.allOf;
+import static com.google.errorprone.matchers.Matchers.hasModifier;
+import static tech.picnic.errorprone.bugpatterns.util.Documentation.BUG_PATTERNS_BASE_URL;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.ErrorProneFlags;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.CompilationUnitTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.CompilationUnitTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.Tree.Kind;
+import com.sun.source.tree.VariableTree;
+import com.sun.source.util.TreeScanner;
+import com.sun.tools.javac.code.Symbol.VarSymbol;
+import java.util.Locale;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import javax.lang.model.element.Modifier;
+import org.jspecify.annotations.Nullable;
+import tech.picnic.errorprone.bugpatterns.util.Flags;
+
+/**
+ * A {@link BugChecker} that warns and suggests the fix when a constant variable does not follow the
+ * upper snake case naming convention.
+ *
+ * <p>Example:
+ *
+ * <p>This checker will re-write the following variables with all its references:
+ *
+ * <ul>
+ *   <li>private static final int number = 1;
+ * </ul>
+ *
+ * <p>To the following:
+ *
+ * <ul>
+ *   <li>private static final int NUMBER = 1;
+ * </ul>
+ *
+ * @apiNote This {@link BugChecker checker} has two optional flags:
+ *     <ul>
+ *       <li>ExcludedConstantFliedNames: A list of field names to exclude from this check.
+ *       <li>IncludePublicConstantFields: Whether to include public constants when running this
+ *           check.
+ *     </ul>
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary =
+        "Make sure that all constant variables follow the `UPPER_SNAKE_CASE` naming convention.",
```

Comment:
```suggestion
        "Make sure that all constant variables follow the `UPPER_SNAKE_CASE` naming convention",
```
Pushing a small suggestion :)